### PR TITLE
Support SECURITY LABEL ON ROLE from any node

### PR DESCRIPTION
**Pull request by eaydingol**
DESCRIPTION: Propagates SECURITY LABEL ON ROLE statement from any node

### CODE REVIEW

File: src/backend/distributed/commands/seclabel.c [hunk 1]
```diff
@@ -11,6 +11,8 @@
 
 #include "postgres.h"
 
+#include "nodes/pg_list.h"
+
```
Comment (onurctirtir) [inline src/backend/distributed/commands/seclabel.c:15]: Do we really need to include this?

File: src/test/regress/sql/seclabel.sql [hunk 2]
```diff
@@ -62,11 +62,20 @@ SET citus.grep_remote_commands = '%SECURITY LABEL%';
 SECURITY LABEL for "citus '!tests_label_provider" ON ROLE user1 IS 'citus_classified';
 SECURITY LABEL ON ROLE user1 IS NULL;
 SECURITY LABEL ON ROLE user1 IS 'citus_unclassified';
-SECURITY LABEL for "citus '!tests_label_provider" ON ROLE "user 2" IS 'citus ''!unclassified';
+SECURITY LABEL for "citus '!tests_label_provider" ON ROLE "user 2" IS 'citus_classified';
```
Comment (onurctirtir) [inline src/test/regress/sql/seclabel.sql:65]: I couldn't quite understand why we had to change the security label specified in an existing test case

File: src/backend/distributed/commands/seclabel.c (hunk 1 above)
Comment (eaydingol) [reply src/backend/distributed/commands/seclabel.c]: You are right, we do not, thanks

File: src/test/regress/sql/seclabel.sql (hunk 2 above)
Comment (eaydingol) [reply src/test/regress/sql/seclabel.sql:65]: We did not have to, it just made easier to update the expected results. The new tests change labels of both users. That security label is applied from the worker in the following lines.

File: src/test/regress/sql/seclabel.sql [hunk 3]
```diff
@@ -62,11 +62,20 @@ SET citus.grep_remote_commands = '%SECURITY LABEL%';
...
+SET citus.grep_remote_commands = '%SECURITY LABEL%';
+-- command from the worker node should be propagated to the master
+SELECT node_type, result FROM get_citus_tests_label_provider_labels('user1') ORDER BY node_type;
+SECURITY LABEL for "citus '!tests_label_provider" ON ROLE user1 IS 'citus_classified';
+SELECT node_type, result FROM get_citus_tests_label_provider_labels('user1') ORDER BY node_type;
+
+RESET citus.log_remote_commands;
+SECURITY LABEL for "citus '!tests_label_provider" ON ROLE "user 2" IS 'citus ''!unclassified';
+SELECT node_type, result FROM get_citus_tests_label_provider_labels('"user 2"') ORDER BY node_type;
+RESET citus.log_remote_commands;
```
Comment (onurctirtir) [inline src/test/regress/sql/seclabel.sql:78]: Given that we already reset this on Line 75, we don't have to do that again.

Changes done by SET/RESET commands persist for the rest of the session, i.e., applicable until the next line that has `\c ..` in regression tests.
```suggestion
```

File: src/test/regress/sql/seclabel.sql [hunk 4]
```diff
@@ -82,6 +91,21 @@ SELECT 1 FROM citus_add_node('localhost', :worker_2_port);
 SELECT node_type, result FROM get_citus_tests_label_provider_labels('user1') ORDER BY node_type;
 SELECT node_type, result FROM get_citus_tests_label_provider_labels('"user 2"') ORDER BY node_type;
 
+-- disable the GUC and check that the command is not propagated
+SET citus.enable_alter_role_propagation TO off;
+SECURITY LABEL ON ROLE user1 IS 'citus_unclassified';
+SELECT node_type, result FROM get_citus_tests_label_provider_labels('user1') ORDER BY node_type;
+RESET citus.enable_alter_role_propagation;
+RESET citus.log_remote_commands;
```
Comment (onurctirtir) [inline src/test/regress/sql/seclabel.sql:99]: we don't have to reset them given that we're anyway switching the session at Line 101

**Review by onurctirtir — APPROVED** [review]

File: src/test/regress/sql/seclabel.sql [hunk 5]
```diff
@@ -62,11 +62,20 @@ SET citus.grep_remote_commands = '%SECURITY LABEL%';
...
 SECURITY LABEL ON ROLE user1 IS 'citus_unclassified';
-SECURITY LABEL for "citus '!tests_label_provider" ON ROLE "user 2" IS 'citus ''!unclassified';
+SECURITY LABEL for "citus '!tests_label_provider" ON ROLE "user 2" IS 'citus_classified';
 
 \c - - - :worker_1_port
--- command not allowed from worker node
-SECURITY LABEL for "citus '!tests_label_provider" ON ROLE user1 IS 'citus ''!unclassified';
+SET citus.log_remote_commands TO on;
+SET citus.grep_remote_commands = '%SECURITY LABEL%';
+-- command from the worker node should be propagated to the master
```
Comment (onurctirtir) [inline src/test/regress/sql/seclabel.sql:70]: We prefer to use "coordinator" instead of "master" in new test files / comments etc.

**Pull request merged**